deepfence_agent/tools/apache/scope/vendor/github.com/weaveworks/tcptracer-bpf/pkg/tracer/tracer.go

```diff
@@ -31,26 +32,46 @@ func TracerAsset() ([]byte, error) {
 	return buf, nil
 }
 
-func NewTracer(cb Callback) (*Tracer, error) {
+func loadBpfModule() (*bpflib.Module, error) {
+	retry := 1
+	var m *bpflib.Module
 	buf, err := Asset("tcptracer-ebpf.o")
 	if err != nil {
 		return nil, fmt.Errorf("couldn't find asset: %s", err)
 	}
-	reader := bytes.NewReader(buf)
+	for {
+		reader := bytes.NewReader(buf)
+		m = bpflib.NewModuleFromReader(reader)
+		if m == nil {
+			return nil, fmt.Errorf("BPF not supported")
+		}
 
-	m := bpflib.NewModuleFromReader(reader)
-	if m == nil {
-		return nil, fmt.Errorf("BPF not supported")
-	}
+		sectionParams := make(map[string]bpflib.SectionParams)
+		sectionParams["maps/tcp_event_ipv4"] = bpflib.SectionParams{PerfRingBufferPageCount: 256}
+		err = m.Load(sectionParams)
+		if err != nil {
+			return nil, err
+		}
 
-	sectionParams := make(map[string]bpflib.SectionParams)
-	sectionParams["maps/tcp_event_ipv4"] = bpflib.SectionParams{PerfRingBufferPageCount: 256}
-	err = m.Load(sectionParams)
-	if err != nil {
-		return nil, err
+		err = m.EnableKprobes(maxActive)
+		if err == nil {
+			break
+		} else {
+			if retry == 0 {
```

Comment:
The retry value is currently set to 1. Which means that this loop will be executed only once ! I think we should set the retry value to a larger number with some wait time between the retries. Or we should remove this piece of code.